examples/trk_subsampler.rs

```diff
@@ -40,18 +44,45 @@ fn main() {
     let mut writer = Writer::new(
         args.get_str("<output>"), Some(reader.header.clone())).unwrap();
 
+    let mut rng = match args.get_str("--seed").parse::<u8>() {
+        Ok(seed) => SmallRng::from_seed([seed; 16]),
+        _        => SmallRng::from_entropy()
+    };
+
     if let Ok(percent) = args.get_str("--percent").parse::<f32>() {
         let percent = percent / 100.0;
-        let mut rng = rand::thread_rng();
 
         for streamline in reader.into_iter() {
             if rng.gen::<f32>() < percent {
                 writer.write(&streamline);
             }
         }
-    } else if let Ok(_nb) = args.get_str("--number").parse::<usize>() {
-        panic!("Not implemented yet");
+    } else if let Ok(nb) = args.get_str("--number").parse::<usize>() {
+        let size = reader.header.nb_streamlines;
+        let mut number = nb;
+
+        if number > size {
+            println!(
+                "The number {} exceed the total number of streamlines: {}. \
+                 Saving {} streamlines.",
+                number, size, size);
+            number = size;
+        } else if number == 0 {
+            panic!("Saving 0 streamline is not usefull. \
+                    Please change the number of steamlines you want.");
```

Comment:
FYI, "usefull" is not a word. Maybe change for this: "You requested a subsampling of 0 streamline. Please ask for any non-zero positive number."